zml/aio/torch.zig

```diff
@@ -314,79 +314,76 @@ pub const PickleData = struct {
-                                    new_prefix.appendAssumeCapacity('.');
+                            }
+                            const used_tag = switch (tag) {
+                                .int => .int64,
+                                .float => .float64,
+                                .bool => .boolval,
+                                else => unreachable,
+                            };
+                            switch (valid_slice) {
+                                true => try store._metadata.put(
+                                    allocator,
+                                    try allocator.dupe(u8, prefix.items),
+                                    .{ .array = .{ .item_type = used_tag, .data = std.mem.sliceAsBytes(try values.toOwnedSlice(allocator)) } },
+                                ),
+                                false => for (values.items, 0..) |val, i| {
+                                    var new_prefix = prefix;
+                                    if (prefix.items.len > 0) {
+                                        new_prefix.appendAssumeCapacity('.');
+                                    }
+                                    new_prefix.items.len += std.fmt.formatIntBuf(new_prefix.unusedCapacitySlice(), i, 10, .lower, .{});
+                                    try store._metadata.put(allocator, try allocator.dupe(u8, new_prefix.items), @unionInit(zml.aio.Value, @tagName(used_tag), val));
+                                },
+                            }
+                        },
+                        else => {
+                            for (seq[1], 0..) |item, i| {
+                                var new_prefix = prefix;
+                                if (v.isPrimitive()) {
+                                    if (prefix.items.len > 0) {
+                                        new_prefix.appendAssumeCapacity('.');
+                                    }
+                                    new_prefix.items.len += std.fmt.formatIntBuf(new_prefix.unusedCapacitySlice(), i, 10, .lower, .{});
                                 }
-                                new_prefix.items.len += std.fmt.formatIntBuf(new_prefix.unusedCapacitySlice(), i, 10, .lower, .{});
+                                try self.parseValue(allocator, store, new_prefix, item);
                             }
-                            try self.parseValue(allocator, store, new_prefix, item);
-                        }
+                        },
                     }
                 },
-                .dict => {
-                    for (seq[1]) |item| {
-                        try self.parseValue(allocator, store, prefix, item);
-                    }
+                .dict => for (seq[1]) |item| {
+                    try self.parseValue(allocator, store, prefix, item);
                 },
                 .kv_tuple => {
                     const key = seq[1][0];
```

Comment:
here also you should use tuple destructuring: `const key, cons val = seq[1]`.

Reply:
Since it's a slice, I had to destructure as such: `const key, const val = seq_values[0..2].*;`. This is fixed in next commit.